packages/extension/src/companion/useCompanionSettings.ts

```diff
@@ -1,35 +1,16 @@
-import { useContext, useEffect, useRef, useMemo } from 'react';
+import { useContext, useEffect, useRef } from 'react';
 import SettingsContext from '@dailydotdev/shared/src/contexts/SettingsContext';
-import FeaturesContext from '@dailydotdev/shared/src/contexts/FeaturesContext';
-import {
-  Features,
-  getFeatureValue,
-} from '@dailydotdev/shared/src/lib/featureManagement';
 import { useExtensionPermission } from './useExtensionPermission';
 
-interface UseCompanionSettings {
-  placement: string;
-}
-
-export const useCompanionSettings = (origin: string): UseCompanionSettings => {
+export const useCompanionSettings = (origin: string): void => {
   const isOnLoad = useRef(true);
-  const { flags } = useContext(FeaturesContext);
-  const placement = getFeatureValue(
-    Features.CompanionPermissionPlacement,
-    flags,
-  );
   const { optOutCompanion, toggleOptOutCompanion, loadedSettings } =
     useContext(SettingsContext);
   const { contentScriptGranted, requestContentScripts } =
     useExtensionPermission({ origin });
 
   useEffect(() => {
-    if (
-      placement === 'off' ||
-      optOutCompanion ||
-      contentScriptGranted ||
-      !loadedSettings
-    ) {
+    if (optOutCompanion || contentScriptGranted || !loadedSettings) {
```

Comment:
You can extract this condition as single variable `shouldToggleCompanion` and then have only that as dependency. I think then we can almost remove the `@NOTE` also.